WatsonWebsocket/WatsonWsClient.cs

```diff
@@ -1,472 +1,813 @@
+        {
+            _IsConnecting = true;
+            _Stats = new Statistics();
+            if (_AcceptInvalidCertificates) ServicePointManager.ServerCertificateValidationCallback += (sender, certificate, chain, sslPolicyErrors) => true;
+            
+            // Pre Configure WebSocket Client
+            _ClientWs.Options.Cookies = _Cookies;
+            _ClientWs.Options.KeepAliveInterval = TimeSpan.FromSeconds(_KeepAliveIntervalSeconds);
+            if (_PreConfigureOptions != null) _PreConfigureOptions(_ClientWs.Options);
+
+            // Connect
+            return _ClientWs.ConnectAsync(_ServerUri, _Token).ContinueWith(AfterConnect);
+        }
+
+        /// <summary>
+        /// Start the client and wait until to connect to the server.
+        /// </summary>
+        /// <param name="token">Cancellation token allowing for termination of this connecting.</param>
+        /// <returns>Boolean indicating if the connection was successfully.</returns>
+        public bool Start(CancellationToken token = default)
+        {
+            _IsConnecting = true;
+            _Stats = new Statistics();
+            if (_AcceptInvalidCertificates) ServicePointManager.ServerCertificateValidationCallback += (sender, certificate, chain, sslPolicyErrors) => true;
+
+            Stopwatch sw = new Stopwatch();
+            TimeSpan timeOut = TimeSpan.FromSeconds(_ConnectTimeoutSeconds);
+            sw.Start();
+
+            while (sw.Elapsed < timeOut)
+            {
+                if (token.IsCancellationRequested) break;
+                // Reset ClientWebSocket (It dispose it-self)
+                _ClientWs = new ClientWebSocket();
+
+                // Pre Configure WebSocket Client
+                _ClientWs.Options.Cookies = _Cookies;
+                _ClientWs.Options.KeepAliveInterval = TimeSpan.FromSeconds(_KeepAliveIntervalSeconds);
+                if (_PreConfigureOptions != null) _PreConfigureOptions(_ClientWs.Options);
+
+                // Try Connect
+                try
+                {
+                    _ClientWs.ConnectAsync(_ServerUri, token).ContinueWith(AfterConnect).Wait();
+                }
+                catch (System.Net.WebSockets.WebSocketException) { }
+                Thread.Sleep(500); // Wait to Reconnect or Change Connecting State
+
+                // Check Connected
+                if (_ClientWs.State == WebSocketState.Open)
+                {
+                    return true;
+                }
+            }
+            _IsConnecting = false;
+            return false;
+        }
+
+        /// <summary>
+        /// Start the client and wait until to connect to the server.
+        /// </summary>
+        /// <param name="token">Cancellation token allowing for termination of this connecting.</param>
+        /// <returns>Task with Boolean indicating if the connection was successfully.</returns>
+        public async Task<bool> StartAsync(CancellationToken token = default)

```

Comment:
In general I think it's a bad idea to have a ```void Start``` and a ```bool Start```, even though the permutations that return ```boolean``` have an optional ```CancellationToken``` parameter.  I'd prefer if they were named differently, i.e. ```Start``` would retain the existing behavior and ```StartWithTimeout``` would be the behavior you're aiming for

Reply:
I thought the same but the name was extremely long but comprehensive